src/components/AdvanceAnalyticsV2/data/hooks/useEnterpriseEnrollmentsData.js

```diff
@@ -0,0 +1,66 @@
+import { useQuery } from '@tanstack/react-query';
+import _ from 'lodash';
+
+import { useMemo } from 'react';
+import { ANALYTICS_TABS, generateKey } from '../constants';
+import { applyGranularity, applyCalculation } from '../utils';
+import EnterpriseDataApiService from '../../../../data/services/EnterpriseDataApiService';
+
+const applyDataTransformations = (response, granularity, calculation, allowedEnrollTypes = ['certificate', 'audit']) => {
+  const modifiedResponse = _.cloneDeep(response);
+  if (modifiedResponse && modifiedResponse.data && modifiedResponse.data.enrollmentsOverTime) {
```

Comment:
[nit] We can simplify this check by using optional chaining `if (modifiedResponse?.data?.enrollmentsOverTime)
`